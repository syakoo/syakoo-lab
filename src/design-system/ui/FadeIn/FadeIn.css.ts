import { keyframes, style } from "@vanilla-extract/css";

const fadeInAnimation = keyframes({
  "0%": {
    opacity: 0,
    transform: "translateY(10px)",
  },
  "100%": {
    opacity: 1,
    transform: "translateY(0)",
  },
});

export const fadeInStyle = style({
  opacity: 0,
  animation: `${fadeInAnimation} forwards cubic-bezier(0.25, 0.1, 0.05, 1.01)`,
});
