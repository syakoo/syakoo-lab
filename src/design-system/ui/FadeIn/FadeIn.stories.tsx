import type { Meta, StoryObj } from "@storybook/react";

import { tokens } from "@/design-system/tokens";

import { FadeIn } from ".";

const meta = {
  component: FadeIn,
  tags: ["autodocs"],
  parameters: { testLevel: "snapshot" },
} satisfies Meta<typeof FadeIn>;

export default meta;

const SampleInnerContent = () => (
  <div
    style={{
      borderRadius: tokens.radii[100],
      background: tokens.colors.palette.gray[100],
      padding: "50px",
      textAlign: "center",
    }}
  >
    Children
  </div>
);

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    children: <SampleInnerContent />,
  },
};
