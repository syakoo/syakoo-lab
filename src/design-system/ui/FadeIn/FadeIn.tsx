import { PolymorphicComponent } from "@/utils/polymorphicComponent";

import { fadeInStyle } from "./FadeIn.css";

type FadeInProps = {
  children: React.ReactNode;
  as?: React.ElementType;
  durationSec?: number;
  delaySec?: number;
};

/**
 * 下からフェードインするアニメーションを提供するラッパーコンポーネント
 */
export const FadeIn: React.FC<FadeInProps> = ({
  children,
  as = "div",
  durationSec = 0.5,
  delaySec = 0,
}) => {
  return (
    <PolymorphicComponent
      as={as}
      className={fadeInStyle}
      style={{
        animationDuration: `${durationSec}s`,
        animationDelay: `${delaySec}s`,
      }}
    >
      {children}
    </PolymorphicComponent>
  );
};
