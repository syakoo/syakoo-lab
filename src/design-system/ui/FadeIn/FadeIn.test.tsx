import { testStories } from "@/test/testStories";

import * as stories from "./FadeIn.stories";

describe("FadeIn", () => {
  testStories(stories);
});
