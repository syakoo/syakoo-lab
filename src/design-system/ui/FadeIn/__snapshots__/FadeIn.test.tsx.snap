// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn スナップショット Sample 1`] = `
<body>
  <div>
    <div
      class="FadeIn_fadeInStyle__r7lc2h1"
      style="animation-duration: 0.5s; animation-delay: 0s;"
    >
      <div
        style="border-radius: 8px; background: rgb(64, 74, 84); padding: 50px; text-align: center;"
      >
        Children
      </div>
    </div>
  </div>
</body>
`;
